### DISCUSSION TITLE
`Grant write access` does not seem to work for markor or gallery

### DISCUSSION THREAD

**Discussion by braindevices** [hardcore-sushi/DroidFS#267] [Q&A]
I enabled `Allow opening files with other applications` and `Grant write access`.
i try to add some text to a file with markor, but this does not really work. markor said file saved, but once I close the markor, the modification goes away indicate the file is never written/saved.

I also try to open photo in the volume by gallery, but it will just crash. I guess even simple modification of the photo file won't work, because normally it requires to write to a new file instead of directly write to the original file.

**Comment by braindevices** [comment]
ok, from https://github.com/hardcore-sushi/DroidFS/issues/54 it looks like, markor use a tmp file to cache the memory file.
However, this does not means it did not save the file through content provider.
Take a look at here https://github.com/gsantner/markor/blob/60d1f434a7fe94ba93fa357754636a8016880bb1/app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java#L2275

Markor do use `context.getContentResolver().openOutputStream(uri, "rwt")` to save it back. So I think it is not necessarily the markor's problem

Can anyone provide an example text editor that can actually write back to droidFS when the features are enabled?

I suspect this never worked.

**Comment by hardcore-sushi** [comment]
Yes you are right. After a quick reading of the markor's code, I think it should work, but I cannot figure out why it doesn't. Nevertheless, I can confirm the feature is working fine with other text editors including [SimpleTextEditor](https://f-droid.org/en/packages/com.maxistar.textpad) as well as [Material Files](https://f-droid.org/en/packages/me.zhanghai.android.files)'s and [Amaze](https://f-droid.org/en/packages/com.amaze.filemanager)'s built-in text editors. However, I cannot get it to work with [Editor](https://f-droid.org/en/packages/org.billthefarmer.editor). The issue looks similar to Markor.

> I also try to open photo in the volume by gallery, but it will just crash.

That may be due to the limitations of memory files. With 1c15f9fac80dcf3473c0341ed8f9d363d4d8bf37, you are able to select disk-based method for all exports. This may fix your issue.

**Reply by braindevices** [reply]
I found a common factor of the non-working editors:
```
getContentResolver().openOutputStream(uri, "rwt"))
```
They use `rwt` instead of `wt`, maybe the resource exposed by droidfs cannot be open/write with `rwt`?
